addons/mass_mailing/static/src/builder/components/company_update_dialog.js

```diff
@@ -0,0 +1,75 @@
+import { Component, onWillStart, useState } from "@odoo/owl";
+import { Dialog } from "@web/core/dialog/dialog";
+import { useService } from "@web/core/utils/hooks";
+
+export class ResCompanyUpdateDialog extends Component {
+    static template = "mass_mailing.CompanyUpdateDialog";
+    static components = { Dialog };
+    static props = {
+        close: Function,
+        resId: Number,
+        fields: { type: Object, optional: true },
+        onRecordSaved: { type: Function, optional: true },
+        onCancel: { type: Function, optional: true },
+    };
+    static defaultProps = {
+        onRecordSaved: () => {},
+        onCancel: () => {},
+    };
+
+    setup() {
+        this.orm = useService("orm");
+        this.state = useState({});
+        onWillStart(async () => {
```

Comment:
Is there a particular reason you need to do this `onWillStart` instead of `setup` ?